### ISSUE TITLE
[Bug]: Pimcore 10: FFMPEG called without fullpath in the save-methode of FFMPEG-adapter

### ISSUE THREAD

**Issue by ascheererEberbach** [pimcore/pimcore#10048]
### Expected behavior

When uploading a video as an assets, a preview image AND a reduced version of the video is created. This should work locally, e.g. in a DDEV/Docker-environment and also in a shared-hosting environment, where FFMPEG is installed via Homebrew in the Linuxbrew-Folder (/usr/home/[[USER]]/.linuxbrew/bin/ffmpeg:). 
FYI: [[USER]] is a placeholder.

### Actual behavior

When adding the fullpath via parameter in the services.yaml, the preview-image is created, but then the asset-video-player is loading over and over again. In the log-directory, for every loading-sequence, a file can be found with the following content: 

`Command: 'ffmpeg' '-i' '/usr/www/users/[[USER]]/[[PROJECT]]/var/tmp/temp-file-611968175debb-b8adc9fe34cb43979e108dccf2292a.mp4' '-ab' '128k' '-vb' '700k' '-filter:v' 'scale=500:trunc(ow/a/2)*2' '-strict' 'experimental' '-f' 'mp4' '-vcodec' 'libx264' '-acodec' 'aac' '-g' '100' '-pix_fmt' 'yuv420p' '-movflags' 'faststart' '-threads' '0' '/usr/www/users/[[]USER]/[[PROJECT]]var/tmp/temp-file-611968176113b-9bc2853163cb97eea2035d308382d4.mp4'

sh: 1: exec: ffmpeg: not found`

Every action to provide the path to the linuxbrew-folder, in which ffmpeg resides, didn't work. I tried via HTACCESS, via startup.php, via path-variable in config.yaml ... 

FYI: [[USER]] and [[PROJECT]] are placeholders.

### Steps to reproduce

I'm very sorry, but this isn't a proper bug-report, as I found the issue already. It resides in Pimcore\Video\Adapter\FFMPEG->save() in row 175, where FFMPEG is called without fullpath. 
` array_unshift($command, 'ffmpeg', '-i', realpath($this->file));`

The solution is, to use the method getFfmpegCli(), which is used in the Adapter in every other case: 

`array_unshift($command, $this->getFfmpegCli(), '-i', realpath($this->file));`

As I didn't develop via Github yet, I didn't dare to create a pull request. But I would be very happy, if someone could do this for me and create a solution for everyone in a shared hosting environment.

Thanks in advance,

Alice

**Comment by roland4432** [comment]
Your suggestion to go with 
```php
array_unshift($command, self::getFfmpegCli(), '-i', realpath($this->file));
```

is already live, but the endless loading is still not fixed by this.

https://user-images.githubusercontent.com/16651237/136170541-3e1a4afe-cbe6-44d9-aad4-34c6dcf72bc4.mp4

Sadly, I cannot recreate the bug in my local dev system, since I get this error:

![image](https://user-images.githubusercontent.com/16651237/136170717-ba4873c9-1539-4019-9bc4-35e7bf0a55c4.png)

**Comment by ascheererEberbach** [comment]
Hi Roland,

did you test FFMPEG on the command-line of your local system?
When I had >your< error, FFMPEG was also killed on the command-line because of a memory-limit which was too small.
After setting it to 1GB locally, I was confrontend with the described one :). 
On the server, we have about 4GB memory-limit. 

Greetings

Alice

**Comment by ascheererEberbach** [comment]
Hi Roland,

look in the var/log-folder. Are there some files with FFMPEG in the name? Take the code inside and paste it to your command-line. Then you have the command which is sent by Pimcore to FFMPEG via PHP. If there aren't files, you have to get the FFMPEG-Adapter to log the command.
It this command gets killed on the command-line, you have to increase the memory-limit. 

Grretings

Alice

**Comment by roland4432** [comment]
Hi,

thank you for the hint. The files in the var/tmp folder are created (<hash>.mp4), but there is no logging in the /var/log/[prod/dev].log. So the command is running and the files are created. Though, the error with the missing templates persist. Therefore I don't think it has something to do with memory limits in my local development system (The dev stage is running with a total of 4GB RAM, my local dev container with 8GB).

![image](https://user-images.githubusercontent.com/16651237/137269418-b77b5fb0-44e1-4e95-9b80-a92637e83581.png)

Since the file missing is the poster image and not the video file, I'd assume it has more to do with thumbnail generation and a missing fallback for the file include. So much for the local system.

And I still have no clue, why I got the mentioned problem on the dev stage, as memory should not be an issue :/

**Comment by Maxara** [comment]
Hello @brusch this is also not working in the official demo.

**Comment by brusch** [comment]
@Maxara see #10629

**Comment by dvesh3** [comment]
Fixed by #10629
